python/ray/tune/trial_executor.py

```diff
@@ -174,10 +192,33 @@ def on_step_end(self, trial_runner):
     def force_reconcilation_on_next_step_end(self):
         pass
 
+    def may_warn_insufficient_resource(self, all_trials):
+        autoscaler_enabled = is_ray_cluster()
+        if not any(trial.status == Trial.RUNNING for trial in all_trials):
+            if self._no_running_trials_since == -1:
+                self._no_running_trials_since = time.monotonic()
+            elif time.monotonic(
+            ) - self._no_running_trials_since > _get_warn_threshold(
+                    autoscaler_enabled):
+                warn_prefix = ("If autoscaler is still scaling up, ignore "
+                               "this message." if autoscaler_enabled else
+                               "Autoscaler is disabled.")
+                logger.warning(
+                    f"{warn_prefix} "
+                    f"Resource is not ready after extended amount of time "
+                    f"without any trials running - please consider if the "
+                    f"allocated resource is not enough.")
```

Comment:
Let's make this a bit more explicit and actionable here, suggestion:

```
No trial is running and no new trial has been started within the last {time} seconds. This could be due to the cluster not having enough resources available to start the next trial. Please check if the requested resources can be fulfilled by your cluster, or will be fulfilled eventually (when using the Ray autoscaler).
```